packages/search-ui/src/actions/setSearchTerm.js

```diff
@@ -30,11 +31,16 @@ export default function setSearchTerm(
   if (refresh) {
     this.debounceManager.runWithDebounce(
       debounce,
+      "_updateSearchResults",
       this._updateSearchResults,
       {
         current: 1,
-        filters: []
+        ...(clearFilters && { filters: [] })
```

Comment:
🎉 Nice and clean. Phew okay... I think that's it! I've looked through all the files at this point. I think most of my comments were primarily side linting / comment / dev clarity type things. The main functionality is amazing, seriously huge kudos for your brainwork there!!